src/pages/AssetMapPage/AMDrawer.tsx

```diff
@@ -113,45 +190,83 @@ export const AMDrawer = () => {
     setShowRank(!showRank);
   };
 
+  if (!epochOptions || amEpochId === undefined) {
+    return <div className={classes.root}></div>;
+  }
+
   return (
-    <Drawer open={open} setOpen={handleSetOpen} Icon={<DnsIcon />} anchorRight>
-      <div className={classes.header}>
-        <h5 className={classes.title}>Active Users</h5>
+    <>
+      <Drawer open={open} setOpen={handleSetOpen}>
         <div className={classes.controls}>
+          <div className={classes.filterHeader}>
+            <Text
+              css={{
+                fontWeight: '$bold',
+                fontSize: '$large',
+                lineHeight: '$short',
+                color: '$headingText',
+              }}
+            >
+              Filters
+            </Text>
+            <PinLeftIcon
+              onClick={() => setOpen(!open)}
+              style={{ color: 'currentColor', cursor: 'pointer' }}
+            />
+          </div>
           {showHiddenFeatures && (
-            <Button
+            <MUButton
               onClick={onRankToggle}
               variant="contained"
               color="primary"
               size="small"
               className={clsx(classes.rank, { [classes.rankOff]: !showRank })}
             >
               <SortIcon />
-            </Button>
+            </MUButton>
+          )}
+          <Select
+            defaultValue={String(amEpochId)}
+            options={epochOptions.map(({ value, label }) => ({
+              value: String(value),
+              label,
+            }))}
+            onValueChange={value => setAmEpochId(Number(value))}
+          />
+          {showHiddenFeatures && (
+            <RedSelect
+              value={metric2}
+              options={metricOptions}
+              onChange={v => setMetric2(v as MetricEnum)}
+            />
           )}
           <ApeAutocomplete
             onChange={setSearch}
             freeSolo
             options={SKILLS}
```

Comment:
@karelianpie we need to filter by  the selected skill here

Reply:
Nice catch @crabsinger, but I did not change that functionality.

I just checked and it seems that it is currently broken, should we open a bug card and fix that bug in there or should we increase the scope of this PR and fix it here? What do you reckon?

Reply:
@crabsinger besides this issue that @sshmm is working on, is there anything else that you feel needs addressing in this PR?

Reply:
@crabsinger filtering by selected skill was fixed by @sshmm, just rebased with main and it's working here too

<img width="1792" alt="Screen Shot 2022-05-22 at 9 19 16 am" src="https://user-images.githubusercontent.com/78794805/169671955-e6f2d7bc-f9bd-4085-8748-b09d37ea6d4e.png">